common/lib/xmodule/xmodule/modulestore/perf_tests/generate_asset_xml.py

```diff
@@ -0,0 +1,228 @@
+    return f
+
+
+def user_id():
+    """
+    Fake user id.
+    """
+    return random.randint(1, 100000000)
+
+
+def versions():
+    """
+    Fake versions.
+    """
+    curr_ver = random.randint(1, 500)
+    prev_ver = curr_ver - 1
+
+    def ver_str(ver):
+        """
+        Version string.
+        """
+        return 'v{}.0'.format(ver)
+    return (ver_str(curr_ver), ver_str(prev_ver))
+
+
+def date_and_time():
+    """
+    Fake date/time.
+    """
+    start_date = datetime.now()
+    time_back = timedelta(seconds=random.randint(0, 473040000))  # 15 year interval
+    return start_date - time_back
+
+
+def contenttype():
+    """
+    Random MIME type.
+    """
+    return random.choice((
+        'image/jpeg',
+        'text/html',
+        'audio/aiff',
+        'video/avi',
+        'text/plain',
+        'application/msword',
+        'application/x-gzip',
+        'application/javascript',
+    ))
+
+
+def generate_random_asset_md():
+    """
+    Generates a single AssetMetadata object with semi-random data.
+    """
+    course_key = CourseKey.from_string('org/course/run')
+    asset_key = course_key.make_asset_key(asset_type(), filename())
+    (curr_version, prev_version) = versions()
+    return AssetMetadata(
+        asset_key,
+        pathname=pathname(),
+        internal_name=filename(),
+        locked=locked(),
+        contenttype=contenttype(),
+        thumbnail=filename(),
+        fields=fields(),
+        curr_version=curr_version,
+        prev_version=prev_version,
+        edited_by=user_id(),
+        edited_by_email='staff@edx.org',
+        edited_on=date_and_time(),
+        created_by=user_id(),
+        created_by_email='staff@edx.org',
+        created_on=date_and_time(),
+    )
+
+
+def make_asset_md(amount):
+    """
+    Make a number of fake AssetMetadata objects.
+    """
+    all_asset_md = []
+    for __ in xrange(amount):
+        all_asset_md.append(generate_random_asset_md())
+    return all_asset_md
+
+
+def make_asset_xml(amount, xml_filename):
+    """
+    Make an XML file filled with fake AssetMetadata.
+    """
+    all_md = make_asset_md(amount)
+    xml_root = lxml.etree.Element("assets")
+    for mdata in all_md:
+        asset_element = lxml.etree.SubElement(xml_root, "asset")
+        mdata.to_xml(asset_element)
+    with open(xml_filename, "w") as xml_file:
+        lxml.etree.ElementTree(xml_root).write(xml_file)
+
+
+def validate_xml(xsd_filename, xml_filename):
+    """
+    Validate a generated XML file against the XSD.
+    """
+    with open(xsd_filename, 'r') as f:
+        schema_root = etree.XML(f.read())
+
+    schema = etree.XMLSchema(schema_root)
+    xmlparser = etree.XMLParser(schema=schema)
+
+    with open(xml_filename, 'r') as f:
+        etree.fromstring(f.read(), xmlparser)
+
+
+def main():
+    """
+    Do it.
+    """
+    usage = "usage: ./generate_asset_xml.py [options]"
+    parser = OptionParser(usage, version="generate_asset_xml 0.1")
```

Comment:
I tend to prefer `argparse` for this sort of thing (I think it's easier to write, and optparse is deprecated in 2.7 [https://docs.python.org/2/library/optparse.html])